guides/stable/3D_interactivity.md

```diff
@@ -0,0 +1,131 @@
+    from a point through the data bounding box.
+
+    Parameters
+    ----------
+    position :
+        the position of the point in nD coordinates. World vs. data
+        is set by the world keyword argument.
+    view_direction : np.ndarray
+        a unit vector giving the direction of the ray in nD coordinates.
+        World vs. data is set by the world keyword argument.
+    dims_displayed :
+        a list of the dimensions currently being displayed in the viewer.
+    world : bool
+        True if the provided coordinates are in world coordinates.
+        Default value is True.
+
+    Returns
+    -------
+    start_point : Optional[np.ndarray]
+        The point on the axis-aligned data bounding box that the cursor click
+        intersects with. This is the point closest to the camera.
+        The point is the full nD coordinates of the layer data.
+        If the click does not intersect the axis-aligned data bounding box,
+        None is returned.
+    end_point : Optional[np.ndarray]
+        The point on the axis-aligned data bounding box that the cursor click
+        intersects with. This is the point farthest from the camera.
+        The point is the full nD coordinates of the layer data.
+        If the click does not intersect the axis-aligned data bounding box,
+        None is returned.
+	"""
+```
+
+## Adding 3D interactivity via mouse events
+Custom 3D interactivity can be added via mouse callbacks. The `layer.get_ray_intersections()` function has been designed to work seamlessly with the napari mouse callback event. You can pass the mouse callback event properties to `layer.get_ray_intersections()` get the `start_point` and `end_point` of where the click line intersects the layer data. 
+
+```
+@layer.mouse_drag_callbacks.append
+def on_click(layer, event):
+	# get the points where the click intersects the 
+    start_point, end_point = layer.get_ray_intersections(
+        position=event.position,
+        view_direction=event.view_direction,
+        dims_displayed=event.dims_displayed,
+        world=True
+    )
+    if (start_point is not None) and (end_point is not None):
+    	# use start_point and end_point to interrogate layer data
+```
+
+For an example implementation, see the [`cursor_ray.py`](https://github.com/napari/napari/blob/master/examples/cursor_ray.py) example.
+
+## Getting the layer data under the cursor
+We are currently implementing convenience methods to the layers (`layer.get_value()`) to get the layer data value underneath the cursor that is "on top" (i.e., closest to `start_point`). Like `layer.get_ray_intersections()`, `layer.get_value()` takes the click position, view direction, dims_displayed in either world or layer coordinates (see `world` argument) as input. Thus, it can be easily integrated into a mouse event callback. Note that `layer.get_value()` returns `None` if the layer is not currently visible. See the docstring below for details.
```

Comment:
```suggestion
There are convenience methods in the layer objects (`layer.get_value()`) to get the layer data value underneath the cursor that is "on top" (i.e., closest to `start_point`). Like `layer.get_ray_intersections()`, `layer.get_value()` takes the click position, view direction, dims_displayed in either world or layer coordinates (see `world` argument) as input. Thus, it can be easily integrated into a mouse event callback. Note that `layer.get_value()` returns `None` if the layer is not currently visible. See the docstring below for details.
```

Reply:
Yeah, maybe we add something like `get_value_from_mouse_event()`? Too long?

Reply:
As with the`get_ray_intersections` function that takes an event, I think the `get_value` one should also be on the layer so that the developer doesn't need to import anything extra to make a mouse callback.